### DISCUSSION TITLE
TR1 source code Aspect Ratio

### DISCUSSION THREAD

**Discussion by ed-kurlyak** [LostArtefacts/TRX#1721] [General]
I looking for applying aspect ratio in TR1 source code, and cannot find. For example we have:

```
void Output_DrawRoom(const int16_t *obj_ptr)
{
    obj_ptr = Output_CalcRoomVertices(obj_ptr);
    obj_ptr = Output_DrawObjectGT4(obj_ptr + 1, *obj_ptr);
    obj_ptr = Output_DrawObjectGT3(obj_ptr + 1, *obj_ptr);
    obj_ptr = Output_DrawRoomSprites(obj_ptr + 1, *obj_ptr);
}
```

In which of those functions applied aspect ratio? For example value 1.7 (for screen 16x9). Where in source code applies aspect ratio? Thanks in advance.

**Comment by oziphantom** [comment]
You don't set the aspect ratio per se, you set the screen resolution which then sets ups the screens aspect ratio. If you wanted to change how the scene is rendered within said window modifying `model_view` inside `GFX_3D_Renderer_Init` by setting the Scale Values in the Matrix would probably achieve it.

**Comment by rr-** [comment]
We'll revamp this in #3332
